src/app/component-library/data-address/data-address-type-select/data-address-type-select-items.ts

```diff
@@ -6,6 +6,13 @@ export const dataAddressTypeSelectItems = (
 ): DataAddressTypeSelectItem[] => {
   const items: DataAddressTypeSelectItem[] = [];
 
+  if (type.startsWith('Datasource')) {
+    items.push({
+      id: 'On-Request',
+      label: '"On Request" Data Offer',
+    });
+  }
```

Comment:
<img width="1768" alt="image" src="https://github.com/user-attachments/assets/96fb354c-6bcd-4929-ae05-22c1669ccf32">

This makes the create data offer form scuffed since both use this data-address-type-select-items thing. Need to find a way to only have this for the dialog